code/modules/reagents/chemistry/reagents/medicine_reagents.dm

```diff
@@ -1612,11 +1612,28 @@
 	REMOVE_TRAIT(L, TRAIT_FEARLESS, type)
 	..()
 
-/datum/reagent/medicine/psicodine/on_mob_life(mob/living/carbon/M)
+/datum/reagent/medicine/psicodine/on_mob_life(mob/living/carbon/M) //0.1 Metabolism at tick rate 1.8 secs
+	if(!data)
+		data = list("misc" = 1)
+	data["misc"]++
 	M.jitteriness = max(0, M.jitteriness-6)
 	M.dizziness = max(0, M.dizziness-6)
 	M.confused = max(0, M.confused-6)
 	M.disgust = max(0, M.disgust-6)
+	if(data["misc"] >= 25) //45 seconds
+		if(is_hivemember(M))
+			for(var/datum/antagonist/hivemind/hive in GLOB.antagonists)
+				if(hive.hivemembers.Find(M.mind))
+					var/mob/living/carbon/C = hive.owner.current.get_real_hivehost()
+					if(C)
+						if(C.mind)
+							to_chat(C, "<span class='assimilator'>We detect a surge of psionic energy from a far away vessel before they disappear from the hive. Whatever happened, there's a good chance they're after us now.</span>")
```

Comment:
```suggestion
	if(data["misc"] >= 25 && is_hivemember(M)) // 25 = 45 seconds
		for(var/datum/antagonist/hivemind/hive in GLOB.antagonists)
			if(hive.hivemembers.Find(M.mind))
				var/mob/living/carbon/C = hive.owner.current.get_real_hivehost()
				if(C?.mind)
					to_chat(C, "<span class='assimilator'>We detect a surge of psionic energy from a far away vessel before they disappear from the hive. Whatever happened, there's a good chance they're after us now.</span>")
```

Reply:
This shouldn't have slipped me, thanks for the review